apps_product_creator/models/odoo_module.py

```diff
@@ -0,0 +1,158 @@
+        "# of Products", compute="_compute_product_qty", store=True
+    )
+
+    def action_view_products(self):
+        action = self.env.ref("product.product_normal_action_sell")
+        result = action.read()[0]
+        product_ids = self.mapped("product_template_id.product_variant_ids").ids
+        # choose the view_mode accordingly
+        if len(product_ids) > 1:
+            result["domain"] = "[('id','in', %s)]" % str(product_ids)
+        elif len(product_ids) == 1:
+            res = self.env.ref("product.product_normal_form_view", False)
+            result["views"] = [(res and res.id or False, "form")]
+            result["res_id"] = product_ids[0]
+        return result
+
+    def action_create_product(self):
+        self._create_product()
+
+    def action_update_product(self):
+        self._update_product()
+
+    def _create_product(self):
+        """
+        Create the product template related to the module in current recordset.
+        :return: product.template recordset
+        """
+        product_obj = self.env["product.template"]
+        products = self.env["product.template"]
+        modules = self.filtered(lambda m: not m.product_template_id)
+        domain = [
+            ("odoo_module_id", "in", modules.ids),
+        ]
+        matching_products = product_obj.search(domain)
+        for odoo_module in modules:
+            product = matching_products.filtered(
+                lambda p: p.odoo_module_id == odoo_module
+            )
+            if not product:
+                if not odoo_module.product_template_id:
+                    product_values = odoo_module._prepare_template()
+                    new_product = product_obj.create(product_values)
+                    odoo_module.write({"product_template_id": new_product.id})
+                    products |= new_product
+        return products
+
+    def _update_product(self):
+        attribute = self.env.ref("apps_product_creator.attribute_odoo_version")
+        attribute_val = self.env["product.attribute.value"]
+        modules = self.filtered("product_template_id")
+        self._update_series_product_attribute_values()
+        for module in modules:
+            series = module.module_version_ids.mapped(
+                "repository_branch_id.organization_serie_id.name"
+            )
+            product = module.product_template_id
+            att_line = product.attribute_line_ids.filtered(
+                lambda a: a.attribute_id.id == attribute.id
+            )
+            att_vals = att_line.mapped("value_ids.name")
+            to_update_vals = list(set(series) - set(att_vals))
+            for to_update_val in to_update_vals:
+                att_val = attribute_val.search(
+                    [("name", "=", to_update_val), ("attribute_id", "=", attribute.id)],
+                    limit=1,
+                )
+                att_line.write({"value_ids": [[4, att_val.id]]})
+        product.create_variant_ids()
```

Comment:
I mean something like... (not tested)
```suggestion
            att_val_ids = attribute_val.search(
                [("name", "in", to_update_vals), ("attribute_id", "=", attribute.id)])
            att_line.write({"value_ids": [[4, record.id] for record in att_val_ids]})
            product.create_variant_ids()
```